doc/source/serve/deployment-graph.md

```diff
@@ -0,0 +1,401 @@
+
++++
+
+### **`bind(*args, **kwargs)`**
+
+Once called on supported ray decorated function or class (@ray.remote, @serve.deployment), generates an IR of type DAGNode that acts as the building block of graph building.
+
++++
+
+#### On function
+
+```bind()``` on function produces a DAGNode that can be exeucted with user input.
+
+```{code-cell} ipython3
+import ray
+from ray import serve
+from ray.serve.pipeline.generate import DeploymentNameGenerator
+
+if ray.is_initialized():
+    serve.shutdown()
+    DeploymentNameGenerator.reset()
+    ray.shutdown()
+
+ray.init(num_cpus=4)
+serve.start()
+
+### Setting up clean ray cluster with serve ###
+
+@serve.deployment
+def preprocessor_with_arg(val):
+    print(val)
+
+# Produces a DeploymentFunctionNode with no bound args.
+func_node = preprocessor_with_arg.bind()
+
+dag_handle = serve.run(func_node)
+# Once executed it will execute with arg value of 2.
+print(ray.get(dag_handle.remote(2)))
+
+"""
+(my_func pid=17060) 2
+"""
+```
+
+#### On class constructor 
+
+**`Class.bind(*args, **kwargs)`** constructs and returns a DAGNode that acts as the instantiated instance of Class, where `*args` and `**kwargs` are used as init args.
+
+#### On class method
+
+Once a class is bound with its init args, its class methods can be directly accessed, called or bound with other args.
+
+```{code-cell} ipython3
+import ray
+from ray import serve
+from ray.serve.pipeline.generate import DeploymentNameGenerator
+
+if ray.is_initialized():
+    serve.shutdown()
+    DeploymentNameGenerator.reset()
+    ray.shutdown()
+
+ray.init(num_cpus=4)
+serve.start()
+
+### Setting up clean ray cluster with serve ###
+
+@serve.deployment
+class Model:
+    def __init__(self, val):
+        self.val = val
+    def get(self):
+        return self.val
+
+# Produces a deployment class node instance initialized with bound args value. 
+class_node = Model.bind(5)
+dag_handle = serve.run(class_node)
+# Access get() class method on bound Model
+print(ray.get(dag_handle.get.remote()))
+"""
+get() returns
+
+5
+"""
+```
+
+### DAGNode as args in bind()
+
+DAGNode can also be passed into other DAGNode in dag binding. In the full example below, ```Combiner``` calls into two instantiations of ```Model``` class, which can be bound and passed into ```Combiner```'s constructor as if we're passing in two regular python class instances.
+
+```python
+m1 = Model.bind(1)
+m2 = Model.bind(2)
+combiner = Combiner.bind(m1, m2)
+```
+
+Similarly, we can also pass and bind upstream DAGNode results that will be resolved upon runtime to downstream DAGNodes, in our example, a `DeploymentMethodNode` that access class method of ```Combiner``` class takes two preprocessing DAGNodes' output as well as part of user input.
+
+```python
+preprocessed_1 = preprocessor.bind(dag_input[0])
```

Comment:
you haven't introduced dag input yet, maybe switch the ordering?

Reply:
the ordering is a bit unnatural to me now ... i started off the example first then positioned all other cells down the line to explain what happened. I think now it makes much more sense to just move it back to top but build it up in the incremental way.